### ISSUE TITLE
Dodgy error message for using a list resource in a list context.

### ISSUE THREAD

**Issue by gtmtech** [hashicorp/terraform#9257]
terraform 0.7.2

Using an aws_security_group resource (incorrectly) like this:

```
resource "aws_security_group" "common" {

...

    # Allow DNS queries to the Management Network
    egress {
        from_port   = 53
        to_port     = 53
        protocol    = "tcp"
        cidr_blocks = "${aws_subnet.management.*.cidr_block}"
    }
```

Will give the misleading error message:

```
* resource 'aws_security_group.common' config: use of the splat ('*') operator must be wrapped in a list declaration
```

So the user might think - ok i'll wrap it in the new list declaration:

```
    egress {
        from_port   = 53
        to_port     = 53
        protocol    = "tcp"
        cidr_blocks = "${list(aws_subnet.management.*.cidr_block)}"
    }
```

Then they will receive this:

```
  * aws_security_group.common: egress.0.cidr_blocks: should be a list
```

So they may then try adding square brackets like this:

```
    egress {
        from_port   = 53
        to_port     = 53
        protocol    = "tcp"
        cidr_blocks = ["${list(aws_subnet.management.*.cidr_block)}"]
    }
```

Then they will receive the VERY CONFUSING error message number 2:

```
* Resource 'aws_security_group.common' does not have attribute 'id' for variable 'aws_security_group.common.id'
* Resource 'aws_security_group.common' does not have attribute 'id' for variable 'aws_security_group.common.id'
* Resource 'aws_security_group.common' does not have attribute 'id' for variable 'aws_security_group.common.id'
* Resource 'aws_security_group.common' does not have attribute 'id' for variable 'aws_security_group.common.id'
* Resource 'aws_security_group.common' does not have attribute 'id' for variable 'aws_security_group.common.id'
* Resource 'aws_security_group.common' does not have attribute 'id' for variable 'aws_security_group.common.id'
* Resource 'aws_security_group.common' does not have attribute 'id' for variable 'aws_security_group.common.id'
* Resource 'aws_security_group.common' does not have attribute 'id' for variable 'aws_security_group.common.id'
* Resource 'aws_security_group.common' does not have attribute 'id' for variable 'aws_security_group.common.id'
* Resource 'aws_security_group.common' does not have attribute 'id' for variable 'aws_security_group.common.id'
* Resource 'aws_security_group.common' does not have attribute 'id' for variable 'aws_security_group.common.id'
* Resource 'aws_security_group.common' does not have attribute 'id' for variable 'aws_security_group.common.id'
```

Which is a real brain scratcher. The solution is of course that it was wrong in the first place and should have been:

```
    egress {
        from_port   = 53
        to_port     = 53
        protocol    = "tcp"
        cidr_blocks = ["${aws_subnet.management.*.cidr_block}"]
    }
```

Which is then fine.

But 2 error messages along the way give the WRONG advice, and label the WRONG problem.

**Comment by mach-kernel** [comment]
This is how I lost over an hour today.

**Comment by mildwonkey** [comment]
Hi folks! I am going to close this issue, as it is resolved in terraform 0.12.
As of terraform 0.12, the redundant list brackets are no longer needed and in fact no longer allowed. The original example code now works as expected.

I also want to mention that the error messages in terraform 0.12 are greatly improved, so if one had redundant brackets in code, they would see this message:

```
Error: Incorrect attribute value type

  on main.tf line 7, in resource "aws_subnet" "management":
   7:   cidr_block = ["${var.cidr_blocks[count.index]}"]

Inappropriate value for attribute "cidr_block": string required.
```

Please don't hesitate to open a new issue if you run across any problems after upgrading to terraform 0.12. Cheers!
